clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -123,7 +133,12 @@ else if (clientAuthConfig.equals("requested"))
     }
 
 
-    private SSLContext createSSLContext() throws GeneralSecurityException, IOException  {
+    private SSLContext createSSLContext()
+        throws GeneralSecurityException, IOException, ClassNotFoundException, IllegalAccessException,
+        InstantiationException {
+        for (String providerClassName : providerClasses) {
+            Security.addProvider((Provider) Class.forName(providerClassName).newInstance());
```

Comment:
can you use the Utils.newInstance here . https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L312